### PR TITLE
Fix cache_all missing ignores

### PR DESCRIPTION
**Pull request by gtrkiller**
Some parameters were left out of #4 so here's a fix for that.

### CODE REVIEW

**Review by gregory-schiano — DISMISSED** [review]
LGTM

File: src/charm.py [hunk 1]
```diff
@@ -156,16 +156,10 @@ def _make_ingress_config(self) -> list:
...
-            # try/except KeyError.
-            if relation:
-                backend_site_name = relation.data[relation.app]["service-hostname"]
-                site = backend_site_name
-        except KeyError:
-            pass
+        relation = self.model.get_relation("ingress-proxy")
+        if relation:
+            backend_site_name = relation.data[relation.app]["service-hostname"]
+            site = backend_site_name
```
Comment (arturo-seijas) [inline src/charm.py:162]: Why not assigning directly `site = relation.data[relation.app]["service-hostname"]` and removing the second assignment?

File: templates/nginx_cfg.tmpl [hunk 2]
```diff
@@ -25,6 +25,7 @@ server {{
         proxy_cache {NGINX_KEYS_ZONE};
         proxy_cache_use_stale {NGINX_CACHE_USE_STALE};
         proxy_cache_valid {NGINX_CACHE_VALID};
+        {NGINX_CACHE_ALL}
```
Comment (arturo-seijas) [inline templates/nginx_cfg.tmpl:28]: Would you need a `;` at the end of the line as above?

Comment (gtrkiller) [reply templates/nginx_cfg.tmpl:28]: the variable has the ; already, if cache_ all is activated an empty line will be in place.

Comment (arturo-seijas) [reply templates/nginx_cfg.tmpl]: Given how it works for the other scenarios, this doesn't seem obvious

File: src/charm.py (hunk 1 above)
Comment (arturo-seijas) [reply src/charm.py:162]: What I meant is that you don't need the variable `backend_site_name` at all. It's only used to reassign its value to `site`

File: src/charm.py [hunk 3]
```diff
@@ -183,19 +175,12 @@ def _make_ingress_config(self) -> list:
...
-        except KeyError:
-            pass
+        relation = self.model.get_relation("ingress-proxy")
         if relation:
+            site = relation.data[relation.app]["service-hostname"]
             svc_name = relation.data[relation.app]["service-name"]
             svc_port = relation.data[relation.app]["service-port"]
-            backend_site_name = relation.data[relation.app]["service-hostname"]
-            site = backend_site_name
+            site = relation.data[relation.app]["service-hostname"]
```
Comment (arturo-seijas) [inline src/charm.py:183]: `backend_site_name`is actually used in this method:
`"NGINX_BACKEND_SITE_NAME": backend_site_name,`

**Review by arturo-seijas — CHANGES_REQUESTED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Review by arturo-seijas — DISMISSED** [review]
LGTM

**Review by jdkandersson — APPROVED** [review]

**Review by merkata — APPROVED** [review]

**Pull request merged**